### PR TITLE
Over-engineer-ify

### PR DESCRIPTION
**Pull request by cirla**
Thought this was a cool script and decided to make a PR. Got carried away and overengineered it. Kept everything Python 2 standard library. Hopefully it all still works.

### CODE REVIEW

File: SecretSanta.py [hunk 1]
```diff
@@ -1,136 +1,164 @@
...
+
+class Provider(Enum):
+    GMAIL = 'gmail'
+
+
+ProviderInfo = collections.namedtuple('ProviderInfo', ['hostname', 'port', 'ssl'])
+
+
+PROVIDER_INFO = {
+    Provider.GMAIL: ProviderInfo('smtp.gmail.com', 465, True),
```
Comment (cirla) [inline SecretSanta.py]: Should be easy to add support for the hip, popular ymail.

File: SecretSanta.py [hunk 2]
```diff
@@ -1,136 +1,164 @@
...
+    parser.add_argument('-m', '--mode', choices=Mode.choices(), default=Mode.TEST)
+    parser.add_argument('-t', '--type', choices=Type.choices(), default=Type.SIMPLE)
+    parser.add_argument('-e', '--email', type=str, help='email address to send from')
+    parser.add_argument('-u', '--username', type=str, required=False,
+                        help='username for email account')
+    parser.add_argument('-p', '--password', type=str, required=False,
+                        help='password for email account')
+    parser.add_argument('-r', '--provider', choices=Provider.choices(), default=Provider.GMAIL,
+                        help='email provider')
+    parser.add_argument('-d', '--due-date', type=parse_date, default=DEFAULT_DUE_DATE)
```
Comment (toriagibbs-zz) [inline SecretSanta.py]: Yo, this is a fancy ass arg parser

**Comment by toriagibbs-zz** [comment]
So you write a bunch of sweet code and don't update the documentation? What do you think this is, Bloomberg?

**Comment by cirla** [comment]
Classic. I'll get some fresh docs going tonight.

**Comment by cirla** [comment]
@toriagibbs how's this?

**Comment by toriagibbs-zz** [comment]
It's beautiful

**Pull request merged**